src/components/achievements.js

```diff
@@ -0,0 +1,58 @@
+import React, { useState } from "react";
+import "./main.css";
+
+const Achievements = () => {
+    const [achievementList, setAchievementList] = useState([]);
+
+    const handleAddClick = (e) => {
+        e.preventDefault();
+        const newAchievement = e.target.elements.newAchievement.value;
```

Comment:
instead of fetching these values on form submit you should fetch & save these at time on input change. 

Same in remaining components as well